### ISSUE TITLE
PINCache 2.2?

### ISSUE THREAD

**Issue by benasher44** [pinterest/PINCache#66]
Hey there! I'm not sure if you all already had plans for other commits going into the next version bump for PINCache. If not, I'd be happy to make a PR with the CHANGELOG.md update and podspec version bump. We're anxious to integrate #63 into an internal pod we have, and unfortunately the CocoaPods podspec dsl for a pod's dependency doesn't allow us to specify a git sha.

Are you all ready for PINCache 2.2?

**Comment by garrettmoon** [comment]
@benasher44 Do it!

**Comment by benasher44** [comment]
@garrettmoon: #67 :)

**Comment by benasher44** [comment]
@garrettmoon thanks for the merge! Can you close this once the new version is pushed to CocoaPods trunk?

**Comment by garrettmoon** [comment]
@benasher44 I'm actually hitting an issue pushing to trunk due to a linter error, we can't build for watchOS. Looks like we need to set PIN_APP_EXTENSIONS when building for watch OS. I don't have time to debug this anymore tonight :(

I'll take a look soon, but if you want to give it a go, feel free to download latest master and `pod spec lint PINCache.podspec --verbose`

**Comment by benasher44** [comment]
ooh okay. i can debug some and follow up

**Comment by benasher44** [comment]
@garrettmoon I went back and tried ```pod spec lint --verbose``` on other tags that have already been submitted, and those failed as well. The issue is that there doesn't appear to be a way to set PIN_APP_EXTENSIONS for when  ```pod spec lint``` goes to build the extension targets (watchos, etc.).

In general, there's another way to design this code such that you don't need PIN_APP_EXTENSIONS. I could put up a PR to discuss it. This would allow allow us to add ```pod spec lint``` to the .travis.yml (assuming travis-ci supports this, which I haven't checked on).

**Comment by garrettmoon** [comment]
That'd be amazing! I'm definitely curious about how we can avoid using PIN_APP_EXTENSIONS!

**Comment by benasher44** [comment]
Alrighty! I'll get a PR up soonish.

**Comment by garrettmoon** [comment]
Thanks @benasher44 !

**Comment by ytok** [comment]
Hey y'all, just FYI linting errors start happening after https://github.com/pinterest/PINCache/pull/45, where  `s.watchos.deployment_target = '2.0'` in the podspec is added.

**Comment by benasher44** [comment]
@garrettmoon @ytok seems like this was broken before then, and it's already affecting existing releases. I've fixed the linting and further watch os issues in #68. Since this isn't a regression, maybe it'd be worth just making sure it lints without the watchos line in the podfile, and then releasing anyway?

**Comment by garrettmoon** [comment]
@benasher44 I agree, however the tool won't submit without linting :/ I'm reluctant to push without tvos support because it would be removing a feature people may be relying on.

**Comment by benasher44** [comment]
@garrettmoon okay I'll try porting my fixes from #68 that just fix the lint issues to a new PR based off of master. That should unblock 2.2

**Comment by benasher44** [comment]
@garrettmoon In #69, I've fixed the build errors, which should unblock the release. Once that gets merged in, I'll rebase #68, since most of these changes I just pulled in from that branch.

**Comment by garrettmoon** [comment]
We (you) did it!

**Comment by benasher44** [comment]
:)

**Comment by garrettmoon** [comment]
Thanks for all your work on this!

**Comment by benasher44** [comment]
Thank you for all of the support on your end as well!
